Client/src/Components/Graph/GraphView.tsx

```diff
@@ -77,24 +78,52 @@ export default function GraphView() {
   }
 
   const onRemoveDataset = (datasetId: number) => {
-    setCompleteDatasets(completeDatasets.filter(dataset => dataset.id !== datasetId))
+    const filteredDataset = completeDatasets.filter(dataset => dataset.id !== datasetId)
+    setCompleteDatasets(filteredDataset)
+    calculateExtremeBoundaries(filteredDataset)
   }
 
   const handleDatasetsSelected = (selectedDatasets: ICompleteDatasetEntity[]) => {
-    let notYetSelectedDatasets: ICompleteDatasetEntity[] = selectedDatasets.filter(selectedDataset => !isInStateAlready(selectedDataset))
+    const notYetSelectedDatasets: ICompleteDatasetEntity[] = selectedDatasets.filter(selectedDataset => !isInStateAlready(selectedDataset))
 
-    let mergedDatasets: ICompleteDatasetEntity[] = [...completeDatasets]
+    const mergedDatasets: ICompleteDatasetEntity[] = [...completeDatasets]
     notYetSelectedDatasets.forEach(dataset => {
       mergedDatasets.push(dataset)
     })
 
     setCompleteDatasets(mergedDatasets)
+    calculateExtremeBoundaries(mergedDatasets)
   }
 
   const isInStateAlready = (dataset: ICompleteDatasetEntity) => {
     return completeDatasets.findIndex(existingDataset => existingDataset.id === dataset.id) != -1
   }
 
+  function calculateExtremeBoundaries(datasets: ICompleteDatasetEntity[]) {
+    let minX = 9000, maxX = 0, minY = 9000, maxY = 0

```

Comment:
isnt it possible to have negative values for Min/Max?

Reply:
Yes, those are just temporary values. Within the actual graph, you can put negative values as long as you are not in the logarithmic scale for that axis.